File: easy/ts/1470.ts
// https://leetcode.com/problems/shuffle-the-array/description/

// Time Complexity: O(n)
// Space Complexity: O(1) output array not considered extra space
function shuffle(nums: number[], n: number): number[] {
  let answer: number[] = [];
  let left: number = 0;
  let right: number = n;

  while (left < n && right < nums.length) {
    answer.push(nums[left]);
    answer.push(nums[right]);
    left++;
    right++;
  }
  return answer;
}

File: easy/ruby/1470.rb
# https://leetcode.com/problems/shuffle-the-array/
# @param {Integer[]} nums
# @param {Integer} n
# @return {Integer[]}
def shuffle(nums, n)
    arr = []

    i = 0
    j = (nums.length / 2).floor

    while j < nums.length
        arr.push(nums[i])
        arr.push(nums[j])
        i +=1
        j+=1
    end
    arr
end